From f8f7e36b75905254d9ee613bc7eaf7104189df02 Mon Sep 17 00:00:00 2001
From: Michael Chang <mchang@suse.com>
Date: Fri, 8 Nov 2024 13:01:07 +0800
Subject: [PATCH] 10_linux: Do not enable BLSCFG on s390-emu

GRUB_ENABLE_BLSCFG may be enabled by some image builders for some reason
without knowing it is not ready for a given platform. Here we force it
to be disabled on s390-emu as it is not ready and has reported issues.

Signed-off-by: Michael Chang <mchang@suse.com>
---
 util/grub.d/10_linux.in | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index f5c80798e..5163b492b 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -108,6 +108,12 @@ cat <<EOF
 EOF
 }
 
+machine=`uname -m`
+# Workaround KIWI sets GRUB_ENABLE_BLSCFG=true in s390-emu
+case "x$machine" in
+    xs390 | xs390x) GRUB_ENABLE_BLSCFG=false ;;
+esac
+
 if [ "x${GRUB_ENABLE_BLSCFG}" = "xtrue" ]; then
   if [ x$dirname = x/ ]; then
     if [ -z "${prepare_root_cache}" ]; then
@@ -247,7 +253,6 @@ EOF
 EOF
 }
 
-machine=`uname -m`
 case "x$machine" in
     xi?86 | xx86_64) klist="/boot/vmlinuz-* /vmlinuz-* /boot/kernel-*" ;;
     xaarch64) klist="/boot/Image-* /Image-* /boot/kernel-*" ;;
-- 
2.47.0

